### PR TITLE
SNOW-1011769: Convenience command 'spcs image-registry url' to get registry url based on current account.

### PR DESCRIPTION
**Pull request by sfc-gh-davwang**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Added a new command under `spcs image-registry` called `url` that prints the image registry url **as long as the current role can view at least one image repository**. This matches the normal workflow where users will create an image repository, then get the image registry url.

**Usage** (error message has been updated but gist is same):
registry_url_both

### CODE REVIEW

File: src/snowflake/cli/plugins/spcs/image_registry/commands.py [hunk 1]
```diff
@@ -19,5 +23,19 @@
...
+@app.command()
+@with_output
+@global_options_with_connection
+def url(**options) -> MessageResult:
+    """Gets the image registry URL for the current account. Must be called from a role that can view at least one image repository in the image registry."""
+    try:
+        return MessageResult(RegistryManager().get_registry_url())
+    except NoRepositoriesViewableError:
+        raise ClickException(
+            "Current role cannot view any image repositories. Please ensure that at least one repository exists in your image registry and use a role with read access to at least one image repository before retrieving registry URL."
```
Comment (sfc-gh-yiwang) [inline src/snowflake/cli/plugins/spcs/image_registry/commands.py:40]: "cannot view" sounds like the role is prohibited to view any repository. Could we rephrase it to sth like
`No image repository found. To get the registry url, please switch to a role with read access to at least one image repository or create a new image repository first`

File: src/snowflake/cli/plugins/spcs/image_registry/manager.py [hunk 2]
```diff
@@ -5,6 +5,14 @@
 import requests
 from click import ClickException
 from snowflake.cli.api.sql_execution import SqlExecutionMixin
+from snowflake.connector.cursor import DictCursor
+
+
+class NoRepositoriesViewableError(ClickException):
+    def __init__(
+        self, msg: str = "Current role does not have view access to any repositories"
```
Comment (sfc-gh-yiwang) [inline src/snowflake/cli/plugins/spcs/image_registry/manager.py:13]: We could not differentiate whether the empty result is due to no read access or no image repository at all. Could we just simply say "No image repository found."

**Review by sfc-gh-yiwang — APPROVED** [review]
LGTM, except nit comments about error message

File: src/snowflake/cli/plugins/spcs/image_registry/manager.py [hunk 3]
```diff
@@ -45,3 +53,13 @@ def login_to_registry(self, repo_url):
...
+
+    def get_registry_url(self):
+        repositories_query = "show image repositories in account"
+        result_set = self._execute_query(repositories_query, cursor_class=DictCursor)
+        results = result_set.fetchall()
+        if len(results) == 0:
+            raise NoRepositoriesViewableError()
+        sample_repository_url = results[0]["repository_url"]
+
+        return "/".join(sample_repository_url.split("/")[:-3])
```
Comment (sfc-gh-davwang) [inline src/snowflake/cli/plugins/spcs/image_registry/manager.py:65]: maybe use urllib.parse or add some validation that this is in the expected format (ends with snowflakecomputing.com)

File: src/snowflake/cli/plugins/spcs/image_registry/manager.py (hunk 2 above)
Comment (sfc-gh-davwang) [reply src/snowflake/cli/plugins/spcs/image_registry/manager.py]: good point, changed error message

File: tests_integration/spcs/test_registry.py [hunk 4]
```diff
@@ -11,3 +14,33 @@ def test_token(runner):
...
+        ["spcs", "image-registry", "url", "--role", test_role]
+    )
+    assert fail_result.exit_code == 1, fail_result.output
+    assert "No image repository found." in fail_result.output
+
+    # role should be able to get registry URL once granted read access to an image repository
+    repo_list_cursor = snowflake_session.execute_string(
+        "show image repositories", cursor_class=DictCursor
+    )
+    expected_repo_url = repo_list_cursor[0].fetchone()["repository_url"]
```
Comment (sfc-gh-astus) [inline tests_integration/spcs/test_registry.py:35]: In `test_utils.py` are functions to extract data from snowflake session result. In this case you could use `row_from_snowflake_session`

Comment (sfc-gh-davwang) [reply tests_integration/spcs/test_registry.py]: done!

**Review by sfc-gh-astus — DISMISSED** [review]

**Review by sfc-gh-astus — APPROVED** [review]
